ecr-login/api/factory.go

```diff
@@ -43,6 +44,14 @@ func (defaultClientFactory DefaultClientFactory) NewClient(region string) Client
 	}
 }
 
+// NewClientWithAWSConfig Create new client with AWS Config
```

Comment:
My only concern the backward compatibility. This will be a breaking change if someone depends on this already. If this is not a concern, I would also prefer to change it to `NewClientWithRegion`